src/Relation/HasManyThrough.php

```diff
@@ -35,23 +37,25 @@ class HasManyThrough extends AbstractRelation implements RelationInterface
      */
     public function getForEntity($entity)
     {
-        // first, find the native values in the through type
-        $native_field = $this->native_field;
-        $native_value = $entity->$native_field;
-        $through_coll = $this->through->getCollectionByField(
-            $this->through_native_field,
-            $native_value
-        );
-
-        // now find the foreign values from the through collection
-        $foreign_values = $through_coll->getFieldValues(
-            $this->through_foreign_field
-        );
+        if (isset($this->through, $this->through_native_field, $this->through_foreign_field)) {
```

Comment:
Is this introduced to fix something ?

Reply:
Yes @harikt, it is currently possible for those fields to be null since they are not required in the constructor, so we check to ensure they are set before accessing properties.